src/appshell/view/appmenumodel.cpp

```diff
@@ -407,6 +407,14 @@ MenuItemList AppMenuModel::makeRecentScoresItems()
         UiAction action;
         action.code = "file-open";
         action.title = TranslatableString::untranslatable(file.displayName(/*includingExtension*/ true));
+
+        bool isCloud = projectConfiguration()->isCloudProject(file.path);
+
+        if (isCloud) {
+            item->setNativeMenuBarIconPath("qrc:/resources/cloud_icon.svg");
```

Comment:
You could try setting this to a real path. Something like [`GlobalConfiguration::appDataPath()`](https://github.com/musescore/MuseScore/blob/22c2e76e73220a40dce5654d58cbacb93cc43786/src/framework/global/internal/globalconfiguration.cpp#L56-L62) `+ "/cloud_icon.svg"`.

You'll need to ensure the icon gets installed to that location. To do that, put the file in [share/icons](https://github.com/musescore/MuseScore/tree/master/share/icons) and add this line in [share/icons/CMakeLists.txt](https://github.com/musescore/MuseScore/blob/22c2e76e73220a40dce5654d58cbacb93cc43786/share/icons/CMakeLists.txt):

```CMake
install(FILES cloud_icon.svg DESTINATION ${Mscore_SHARE_NAME}${Mscore_INSTALL_NAME})
```

You can do this on all platforms. On macOS, this location is [inside the `.app` bundle](https://github.com/musescore/MuseScore/blob/22c2e76e73220a40dce5654d58cbacb93cc43786/src/framework/global/internal/globalconfiguration.cpp#L70). You can look inside the `.app` to check that the file ends up in the proper location, and use a print/debug/log statement in the QML/C++ code to make that the path is correct.

Reply:
If it doesn't work with the SVG, try using the PNG instead.

Reply:
I did the changes and tried it for both svg & png, it is not rendering. I checked the path as well and can see for example .png under .app/Resources/
Reference: 
![Screenshot 2024-03-24 at 11 05 38 PM](https://github.com/musescore/MuseScore/assets/18035782/e7e61e2d-484e-467a-9e5c-3800f944ab78)

Reply:
Please push the new code. You can keep the existing code as well for the time being, though you'll need to pick one of the paths and put the other in a comment, like this:

```C++
// item->setNativeMenuBarIconPath("qrc:/resources/cloud_icon.svg");
item->setNativeMenuBarIconPath(GlobalConfiguration::appDataPath() + "/cloud_icon.svg");
```

Reply:
Yeah sure. Pushed the new code. Currently kept the older svg and appshell.qrc update. Once, we finalise on the approach, will accordingly update it.

Reply:
Could you add the [PNG file](https://github.com/musescore/MuseScore/pull/21802#issuecomment-1991931056) as well, and the code to install it?

Reply:
Added and updated the PR